bcgov/common-hosted-form-service
# SSO - Digital Credential Demo

# Description
This is a demo-only PR. This builds on the SSO/IDP changes in #1258 and adds a new IDP - Verified Email.

The SSO Standard realm supports the Verified Email credential in their sandbox environment, so this PR/deployment will make use of a temporary config map (chefs-sandbox-oidc-config) and use `SANDBOX_OIDC_*` environment variables to set up Keycloak/OIDC.

Management of the OIDC client is here: https://bcgov.github.io/sso-requests-sandbox

To test this out, you will need:

- BC Wallet application: https://www2.gov.bc.ca/gov/content/governments/government-id/bc-wallet
- register for verified email: https://email-verification.vonx.io/

1. Download the app and install.
2. Follow the registration procedures, use a valid email you have access to as email is validated and invitation is sent to that address.
3. Perform the acceptance of the invitation on your phone where your BC wallet application is. The email should prompt to open the BC wallet.


The Verified Email credential/login will act like `BCeID Basic` - extremely limited permissions in the app.

Here is a link to a form that requires Verified Email login: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-1311/form/submit?f=e5318071-b2d0-470e-b1df-d0a8c52da208





## Types of changes

- Add new IDP (database migration).
- Add support for additional login parameters.








## Checklist




- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have approval from the product owner for the contribution in this pull request

## Further comments
